### DISCUSSION TITLE
File Structure Overview for Python Files

### DISCUSSION THREAD

**Discussion by vsfeedback** [Microsoft/PTVS#7629] [Enchancements]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/File-Structure-Overview-for-Python-Files/10347715)._

---
It would be great to have a feature like the "Class View" (which is available for C# etc.) also for Python files. PyCharm has a so called "Structure" view, which lists all definitions in the currently open file (as shown in the screenshot below). This helps a lot with navigating through larger files.

![Screenshot 2023-04-25 093314.png](https://aka.ms/dc/image?name=B7ed7f7f814064ed28336be8a90fca015638180049442053634_Screenshot_2023-04-25_093314.png&tid=7ed7f7f814064ed28336be8a90fca015638180049442053634)

---
#### Feedback Bot on 4/25/2023, 10:19 PM: 

(private comment, text removed)

**Comment by StellaHuang95** [comment]
This is a dupe of https://github.com/microsoft/PTVS/issues/6551. I will mark it as an enhancement.

**Comment by chrisstr** [comment]
I'm not sure whether #6551 is a duplicate. I was rather thinking of a separate view for Python files as sidebar, like the "Class View" for C# or the "Structure" view in PyCharm.

<img width="871" alt="B7ed7f7f814064ed28336be8a90fca015638180049442053634_Screenshot_2023-04-25_093314" src="https://github.com/microsoft/PTVS/assets/134502/65998135-e988-473b-a456-0677b7a3dd86">


It might not be different functionally, but I think it's a big difference in usability.

**Comment by StellaHuang95** [comment]
Thanks for the clarification, I will move this issue to discussion as an enhancement request for comments and upvotes.
